### DISCUSSION TITLE
How do you mix useQuery with useAsyncData in Nuxt ?

### DISCUSSION THREAD

**Discussion by liorchamla** [posva/pinia-colada#130] [Questions] [answered]
Ok folks, the library is CRAZY cool ! Thank you for this amazing job, this is really nice !

The only thing I can't get is how do you mix it with `useAsyncData` inside a Nuxt page ? Since the `useQuery` composable fires the query immediately after it is created, you can not decide to launch it and wait for it inside a `useAsyncData` to return the data that resolve after the first query.

So how in the world to you do this thing ?

Again, thank you for this piece of art of library :)

**Comment by posva** [comment, marked as answer]
Thanks! I'm glad you like it

The answer is **you don't**! `useQuery()` has every feature `useAsyncData()` has, so you don't need it.

Why do you want to fire the refresh later on? You can do so with the `enabled` option:

```ts
const enabled = ref(false)
const { data } = useQuery({
  // ...
  enabled,
})

enabled.value = true // let it trigger
```

You can also directly call `refresh()` and that can be awaited

**Reply by liorchamla** [reply]
1. This is mindblowing. You did an AMAZING work here.
2. It remains something I can't figure out : normally, when you activate a NuxtLink, if you target page has a `useAsyncData` inside, the target page won't be displayed while you wait for that data (and also it would trigger NuxtLoadingIndicator as a side effect). How do you reach that look and feel with `useQuery` (specially when this useQuery is inside a Pinia store) ? 

From what I see right now, rather than waiting for the data to be here to display the page, we should display the page with Skeletons inside, before the data arrives.

**Reply by posva** [reply]
You are too kind 😄 

The key is the `await` in front of `useAsyncData()`. Here it must be done in two steps:

```ts
const { data, refresh } = useQuery({
  // ...
})
await refresh()
```

The refresh call dedupes requests and takes into account the cache so it won't be a wasted call. This kind of usage makes sense with Suspense. See https://github.com/posva/pinia-colada/issues/127#issuecomment-2524148104

**Reply by liorchamla** [reply]
THIS ! THIS IS WHAT I WANTED ! I though refresh would only make sense when we need to refresh, but here it seems that if we call it right away, `useQuery` + `await refresh()` still trigger only one HTTP Request !

Incredible. Thank you so much, you are so going to my next production Nuxt 3 app :D :D :D

**Reply by posva** [reply]
Nice! Let me know if you have other feedback
